superset/connectors/sqla/models.py

```diff
@@ -1144,9 +1144,10 @@ def adhoc_column_to_sqla(
             and (time_grain := col.get("timeGrain"))
         ):
             sqla_column = self.db_engine_spec.get_timestamp_expr(
-                sqla_column,
-                None,
-                time_grain,
+                col=sqla_column,
+                pdf=None,
+                time_grain=time_grain,
+                type_=str(getattr(sqla_column, "type", "")),
```

Comment:
`_date_trunc_functions` expects to get string object as `_type`.

Reply:
Could we refactor this so that `type_` is removed from the `BaseEngineSpec.get_timestamp_expr` sig and we then use `sqla_column.type` inside the method for `type_`? Or is the explicit `type_` needed somewhere else? I looked here, and I think we can infer the type from `ColumnClause.type` here, too.